Azure/azure-sdk-for-sap-odata
# Add-apikey

## Purpose
Introduces API Key handling and solves #21

## Does this solve an existing issue?

```
[x] Yes - Please add the reference: 
[ ] No
```

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code

```
Use API Key protected OData service
maintain API Key and respective header name in appsettings.json
verify additional header gets added for instance checking azure-sdk-for-sap-odata\Dependencies\DataOperations.OData\Client\ODataHttpClientContext.cs
```

## What to Check

* request authenticated with API key and potentially additional auth scheme like Basic Auth
